Pod/Classes/Kits/Kahuna/MPKitKahuna.m

```diff
@@ -332,13 +337,32 @@ - (MPKitExecStatus *)setUserIdentity:(NSString *)identityString identityType:(MP
     }
     
     if (identityString) {
-        [KahunaAnalytics setUserCredentialsWithKey:kahunaCredential andValue:identityString];
+        KahunaUserCredentials *kuc = [Kahuna getUserCredentials];
+        [kuc addCredential:kahunaCredential withValue:identityString];
+        NSError *error = nil;
+        [Kahuna loginWithCredentials:kuc error:&error];
+        if (error) {
+            MPLogDebug(@"Kahuna Login Error : %@", error.description);
+        }
     }
     
     execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceKahuna) returnCode:MPKitReturnCodeSuccess];
     return execStatus;
 }
 
+- (void) trackKahunaEvent:(NSString*)eventName withCount:(int)count withValue:(int)value withProperties:(NSDictionary*) properties {
+    if (!eventName) return;
+    
+    KAHEventBuilder *builder = [KAHEventBuilder eventWithName:eventName];
+    [builder setPurchaseCount:count andPurchaseValue:value];
+    for (NSString* eachKey in properties) {
+        NSString *eachValue = properties[eachKey];
+        [builder addProperty:eachKey withValue:eachValue];
```

Comment:
does mParticle have special rules regarding empty key/ empty value for Event properties?